core/src/main/scala/tamer/Setup.scala

```diff
@@ -1,48 +1,12 @@
 package tamer
 
-import com.sksamuel.avro4s._
-import doobie.util.query.Query0
 import tamer.registry.{Registry, Topic}
-import zio.UIO
 import zio.kafka.serde.Serializer
 
-final case class ResultMetadata(queryExecutionTime: Long)
-final case class QueryResult[V](metadata: ResultMetadata, results: List[V])
-
-final case class Setup[K, V, State](
-    keySerializer: Serializer[Registry with Topic, K],
-    valueSerializer: Serializer[Registry with Topic, V],
-    stateSerde: ZSerde[Registry with Topic, State],
-    valueToKey: V => K,
-    defaultState: State,
-    buildQuery: State => Query0[V],
-    stateFoldM: State => QueryResult[V] => UIO[State]
+abstract class Setup[-K, -V, S](
+    val keySerializer: Serializer[Registry with Topic, K],
+    val valueSerializer: Serializer[Registry with Topic, V],
+    val stateSerde: ZSerde[Registry with Topic, S],
+    val defaultState: S,
+    val queryHash: Int
```

Comment:
```suggestion
    val stateKey: String
```
The fact that for the DB we will hash the query and the default state is an implementation detail